src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java

```diff
@@ -140,16 +165,199 @@ private int calculateTotal(int base, TFloatList multipliers, TIntList modifiers)
 
     }
 
-    /** Handles DoHeal event to increase health of entity */
+    /**
+     * Handles DoHeal event to increase health of entity.
+     *
+     * @param event DoHealEvent which triggered the heal effect.
+     * @param entity The entity which will be healed by the event.
+     */
     @ReceiveEvent(components = {HealthComponent.class})
     public void onHeal(DoHealEvent event, EntityRef entity) {
         checkHeal(entity, event.getAmount(), event.getInstigator());
     }
 
-    /** Resets the health of player to maxHealth on re-spawn. */
+    /**
+     * Resets the health of player to maxHealth on re-spawn.
+     *
+     * @param event OnPlayerRespawnedEvent sent during the re-spwan of player.
+     * @param entity The Player entity.
+     * @param healthComponent The health component attached to player entity.
+     */
     @ReceiveEvent
     public void onRespawn(OnPlayerRespawnedEvent event, EntityRef entity, HealthComponent healthComponent) {
         healthComponent.currentHealth = healthComponent.maxHealth;
         entity.saveComponent(healthComponent);
     }
+
+
+    /**
+     * Override the default behavior for an attack, causing it damage as opposed to just destroying it or doing nothing.
+     *
+     * @param event Attack event sent on targetEntity.
+     * @param targetEntity The entity which is attacked.
+     */
+    @ReceiveEvent(components = HealthComponent.class, netFilter = RegisterMode.AUTHORITY)
+    public void onAttackEntity(AttackEvent event, EntityRef targetEntity) {
+        damageEntity(event, targetEntity);
+    }
+
+    static void damageEntity(AttackEvent event, EntityRef targetEntity) {
```

Comment:
Why is this a separate static method to the above?

Reply:
I couldn't find any other methods using `damageEntity` method. So will move it into the `onAttackEntity` method.

Reply:
I did not check in the original Core/Health directory for usage. Turns out `BlockDamageAuthoritySystem` uses this method. [See here](https://github.com/e-aakash/Health/pull/1/files#diff-1a42c6b21c057e315502d39b4f22a5f6R225)
So I will have to either revert the commit or add it back in the next commit.